### PR TITLE
Sample_id and latent vars

### PR DESCRIPTION
**Pull request by wcbeard**
Uses updated functionality from bgbb_lib 0.1.4 to compute and write estimates for latent variables, as well as sample_id. Column names are also changed to be more user friendly, and more diagnostic info is printed out.

Logs from test runs:

fit
https://pastebin.com/6wzcgusJ

pred
https://pastebin.com/WvYza4vz

### CODE REVIEW

File: bgbb_airflow/pred_airflow_job.py [hunk 1]
```diff
@@ -1,50 +1,70 @@
...
 
 
-def pull_most_recent_params(spark, param_bucket, param_prefix):
+def pull_most_recent_params(spark, max_sub_date, param_bucket, param_prefix):
     pars_loc = join(param_bucket, param_prefix)
     print("Reading params from {}".format(pars_loc))
     spars = spark.read.parquet(pars_loc)
-    pars_df = spars.orderBy(spars.submission_date_s3.desc()).limit(1).toPandas()
+    pars_df = (
+        spars.filter(spars.submission_date_s3 <= max_sub_date)
```
Comment (acmiyaguchi) [inline bgbb_airflow/pred_airflow_job.py:32]: `submission_day_s3` is `YYYY-MM-DD` or `YYYYMMDD`?

**Review by acmiyaguchi — APPROVED** [review]

File: bgbb_airflow/pred_airflow_job.py (hunk 1 above)
Comment (wcbeard) [reply bgbb_airflow/pred_airflow_job.py:32]: `YYYY-MM-DD`. I annotated it as a `Dash_str` type [here below](https://github.com/wcbeard/bgbb_airflow/pull/2/files#diff-a6f1f032764b436f766c42b0d3a66971R40). Should I annotate in this arg as well, or is there a better way to specify this?

Comment (acmiyaguchi) [reply bgbb_airflow/pred_airflow_job.py:32]: I think the most human friendly way is to add the expected date format as a help string, but after re-reading, this looks fine.

Comment (wcbeard) [reply bgbb_airflow/pred_airflow_job.py:32]: Ok, I added a help string in commit below

**Pull request merged**